django/db/models/fields/__init__.py

```diff
@@ -782,12 +782,11 @@ def contribute_to_class(self, cls, name, private_only=False):
         self.set_attributes_from_name(name)
         self.model = cls
         cls._meta.add_field(self, private=private_only)
-        if self.column:
+        if self.column and not getattr(cls, self.attname, None):
             # Don't override classmethods with the descriptor. This means that
             # if you have a classmethod and a field with the same name, then
             # such fields can't be deferred (we don't have a check for this).
-            if not getattr(cls, self.attname, None):
-                setattr(cls, self.attname, self.descriptor_class(self))
+            setattr(cls, self.attname, self.descriptor_class(self))
```

Comment:
Above comment doesn't match to the line anymore. Moreover this code is really complicated and fragile, please revert this change.